File: backend/src/test/java/info/gamewise/lor/videos/DataLoader.java
package info.gamewise.lor.videos;

import com.github.brunosc.fetcher.domain.VideoDetails;
import com.github.brunosc.lor.domain.LoRRegion;
import com.google.api.client.util.DateTime;
import com.google.api.services.youtube.model.PlaylistItem;
import com.google.api.services.youtube.model.PlaylistItemSnippet;
import com.google.api.services.youtube.model.ResourceId;
import com.google.api.services.youtube.model.ThumbnailDetails;
import info.gamewise.lor.videos.domain.LoRVideo;
import info.gamewise.lor.videos.domain.VideoChampion;
import info.gamewise.lor.videos.domain.VideoChannel;
import info.gamewise.lor.videos.domain.VideoRegion;
import info.gamewise.lor.videos.domain.json.Champion;
import info.gamewise.lor.videos.domain.json.Channel;
import info.gamewise.lor.videos.port.out.VideosNotInDatabasePort.NewVideo;

import java.time.LocalDate;
import java.util.Set;

import static java.util.stream.Collectors.toUnmodifiableSet;

public class DataLoader {

    public static final Channel MEGA_MOGWAI = new Channel("MEGA_MOGWAI", "MegaMogwai", "1");
    public static final Channel ALANZQ = new Channel("ALANZQ", "Alanzq", "2");
    public static final Channel SILVERFUSE = new Channel("SILVERFUSE", "Silverfuse", "2");

    public static final Champion EZREAL = new Champion("01PZ036", "EZREAL", "EZREAL");
    public static final Champion RENEKTON = new Champion("04SH067", "RENEKTON", "RENEKTON");
    public static final Champion LUCIAN = new Champion("01DE022", "LUCIAN", "LUCIAN");
    public static final Champion HECARIM = new Champion("01SI042", "HECARIM", "HECARIM");
    public static final Champion SWAIN = new Champion("02NX007", "SWAIN", "SWAIN");
    public static final Champion LISSANDRA = new Champion("04FR005", "LISSANDRA", "LISSANDRA");
    public static final Champion SHEN = new Champion("01IO032", "SHEN", "SHEN");
    public static final Champion JARVAN_IV = new Champion("04DE008", "JARVAN_IV", "JARVAN_IV");
    public static final Champion TRUNDLE = new Champion("03FR006", "TRUNDLE", "TRUNDLE");
    public static final Champion IRELIA = new Champion("04IO005", "IRELIA", "IRELIA");
    public static final Champion AZIR = new Champion("04SH003", "AZIR", "AZIR");
    public static final Champion ANIVIA = new Champion("01FR024", "ANIVIA", "ANIVIA");
    public static final Champion ELISE = new Champion("01SI053", "ELISE", "ELISE");
    public static final Champion KALISTA = new Champion("01SI030", "KALISTA", "KALISTA");
    public static final Champion TALIYAH = new Champion("04SH073", "TALIYAH", "TALIYAH");
    public static final Champion MALPHITE = new Champion("04MT008", "MALPHITE", "MALPHITE");
    public static final Champion AURELION_SOL = new Champion("03MT087", "AURELION_SOL", "AURELION_SOL");
    public static final Champion GAREN = new Champion("01DE012", "GAREN", "GAREN");
    
    public DataLoader(){}

    public static VideoDetails videoDetails(String id, String deckCode) {
        return new VideoDetails(playlistItem(id, deckCode, LocalDate.now().toEpochDay()));
    }

    public static VideoDetails videoDetails(String id, String deckCode, long publishedAt) {
        return new VideoDetails(playlistItem(id, deckCode, publishedAt));
    }

    public static NewVideo newVideo(int day, String id, Channel channel, Set<LoRRegion> regions, Set<Champion> champions) {
        return new NewVideo("DECK_CODE", videoDetails(id, "DECK_CODE", LocalDate.of(2021, 1, day).toEpochDay()), channel, regions, champions);
    }

    public static LoRVideo newLoRVideo(Channel channel, Set<Champion> champions, Set<LoRRegion> regions, int day) {
        return new LoRVideo(
                "url " + day,
                "title " + day,
                "CODE" + day,
                new VideoChannel(channel),
                champions.stream().map(VideoChampion::new).collect(toUnmodifiableSet()),
                regions.stream().map(VideoRegion::new).collect(toUnmodifiableSet())        ,
                LocalDate.of(2020, 1, day).atStartOfDay(),
                null
        );
    }

    private static PlaylistItem playlistItem(String id, String deckCode, long publishedAt) {
        PlaylistItemSnippet snippet = new PlaylistItemSnippet()
                .setTitle("Title " + id)
                .setDescription(deckCode)
                .setResourceId(new ResourceId().setVideoId(id))
                .setThumbnails(new ThumbnailDetails())
                .setPublishedAt(new DateTime(publishedAt));

        return new PlaylistItem().setSnippet(snippet);
    }

}

File: backend/src/test/kotlin/info/gamewise/lor/videos/DataLoader.kt
package info.gamewise.lor.videos

import com.github.brunosc.fetcher.domain.VideoDetails
import com.github.brunosc.fetcher.domain.VideoThumbnails.ThumbnailItem
import com.github.brunosc.lor.domain.LoRRegion
import com.google.api.client.util.DateTime
import com.google.api.services.youtube.model.*
import info.gamewise.lor.videos.domain.LoRVideo
import info.gamewise.lor.videos.domain.VideoChampion
import info.gamewise.lor.videos.domain.VideoChannel
import info.gamewise.lor.videos.domain.VideoRegion
import info.gamewise.lor.videos.domain.json.Champion
import info.gamewise.lor.videos.domain.json.Channel
import info.gamewise.lor.videos.port.out.NewVideo
import java.time.LocalDate
import java.util.stream.Collectors

object DataLoader {
    const val DECK_CODE = "CICACAQDAMBQCBJHGU4AGAYFAMCAMBABAMBA6KBXAMAQCAZFAEBAGBABAMCQEAIBAEBS4"

    val MEGA_MOGWAI = Channel("MEGA_MOGWAI", "MegaMogwai", "1")
    val ALANZQ = Channel("ALANZQ", "Alanzq", "2")
    val SILVERFUSE = Channel("SILVERFUSE", "Silverfuse", "2")

    val EZREAL: Champion = Champion("01PZ036", "EZREAL", "EZREAL")
    val RENEKTON: Champion = Champion("04SH067", "RENEKTON", "RENEKTON")
    val LUCIAN: Champion = Champion("01DE022", "LUCIAN", "LUCIAN")
    val HECARIM: Champion = Champion("01SI042", "HECARIM", "HECARIM")
    val SWAIN: Champion = Champion("02NX007", "SWAIN", "SWAIN")
    val LISSANDRA: Champion = Champion("04FR005", "LISSANDRA", "LISSANDRA")
    val SHEN: Champion = Champion("01IO032", "SHEN", "SHEN")
    val JARVAN_IV: Champion = Champion("04DE008", "JARVAN_IV", "JARVAN_IV")
    val TRUNDLE: Champion = Champion("03FR006", "TRUNDLE", "TRUNDLE")
    val IRELIA: Champion = Champion("04IO005", "IRELIA", "IRELIA")
    val AZIR: Champion = Champion("04SH003", "AZIR", "AZIR")
    val ANIVIA: Champion = Champion("01FR024", "ANIVIA", "ANIVIA")
    val ELISE: Champion = Champion("01SI053", "ELISE", "ELISE")
    val KALISTA: Champion = Champion("01SI030", "KALISTA", "KALISTA")
    val TALIYAH: Champion = Champion("04SH073", "TALIYAH", "TALIYAH")
    val MALPHITE: Champion = Champion("04MT008", "MALPHITE", "MALPHITE")
    val AURELION_SOL: Champion = Champion("03MT087", "AURELION_SOL", "AURELION_SOL")
    val GAREN: Champion = Champion("01DE012", "GAREN", "GAREN")
    val TEEMO: Champion = Champion("01PZ008", "TEEMO", "TEEMO")
    val NASUS: Champion = Champion("04SH047", "NASUS", "NASUS")
    val THRESH: Champion = Champion("01SI052", "THRESH", "THRESH")
    val NAMI: Champion = Champion("05BW005", "NAMI", "NAMI")
    val TWISTED_FATE: Champion = Champion("02BW026", "TWISTED_FATE", "TWISTED_FATE")

    @JvmStatic
    fun videoDetails(id: String, deckCode: String): VideoDetails {
        return VideoDetails(playlistItem(id, deckCode, LocalDate.now().toEpochDay()))
    }

    @JvmStatic
    fun videoDetails(id: String, deckCode: String, publishedAt: Long): VideoDetails {
        return VideoDetails(playlistItem(id, deckCode, publishedAt))
    }

    fun newVideo(
        day: Int,
        id: String,
        channel: Channel,
        regions: Set<LoRRegion>,
        champions: Set<Champion>
    ): NewVideo {
        return NewVideo(
            "DECK_CODE",
            videoDetails(id, "DECK_CODE", LocalDate.of(2021, 1, day).toEpochDay()),
            channel,
            regions,
            champions
        )
    }

    @JvmStatic
    fun newLoRVideo(channel: Channel, champions: Set<Champion>, regions: Set<LoRRegion>, day: Int): LoRVideo {
        return LoRVideo(
            url = "url $day",
            title = "title $day",
            deckCode = "CODE$day",
            channel = VideoChannel(channel),
            champions = champions.stream()
                .map { champion: Champion -> VideoChampion(champion) }
                .collect(Collectors.toUnmodifiableSet()),
            regions = regions.stream().map { region: LoRRegion -> VideoRegion(region) }
                .collect(Collectors.toUnmodifiableSet()),
            publishedAt = LocalDate.of(2020, 1, day).atStartOfDay(),
            thumbnail = ThumbnailItem()
        )
    }

    private fun playlistItem(id: String, deckCode: String, publishedAt: Long): PlaylistItem {
        val thumbnail = ThumbnailDetails().setHigh(Thumbnail().setHeight(200).setWidth(200))

        val snippet: PlaylistItemSnippet = PlaylistItemSnippet()
            .setTitle("Title $id")
            .setDescription(deckCode)
            .setResourceId(ResourceId().setVideoId(id))
            .setThumbnails(thumbnail)
            .setPublishedAt(DateTime(publishedAt))
        return PlaylistItem().setSnippet(snippet)
    }
}